v2/internal/testcommon/samples_tester.go

```diff
@@ -48,6 +48,9 @@ var exclusions = []string{
 
 	// Excluding dbformysql/user as is not an ARM resource
 	"user",
+
+	// Excluding sqlroleassignment because it needs to reference a principalId
+	"sqlroleassignment",
```

Comment:
can this be done the same way we do `roleassignment`?

Reply:
It could but since that requires hardcoding of a principalID or editing of the sample before re-recording I decided not to do it and instead just bypass this test for now. I'm got a proposal out that will make this easier because we can then link via the UMI name. Once that's implemented we can run this sample.